pkg/concurrent/concurrent.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package concurrent
+
+import (
+	"sync"
+
+	"sigs.k8s.io/kind/pkg/util"
+)
+
+// UntilError runs all funcs in separate goroutines, returning the
+// first non-nil error returned from funcs, or nil if all funcs return nil
+func UntilError(funcs []func() error) error {
+	errCh := make(chan error, len(funcs))
+	for _, f := range funcs {
+		f := f // capture f
+		go func() {
+			errCh <- f()
+		}()
+	}
+	for i := 0; i < len(funcs); i++ {
```

Comment:
just curious, this `for` loop is the one that waits for all goroutines to finish, isn´t it?

Reply:
UntilError returns the first error immediately without waiting.

Reply:
ic, and if no errors it waits for all functions to send the err results. Thanks

Reply:
right, we can't return `nil` correctly until every func has returned, but we can return an error immediately, this one is meant for quick bail-out (EG for cluster creation code), while we have `Coalesce` for when we want to continue and collect the errors (as in collecting logs).